src/Symfony/Bridge/Twig/Resources/views/Form/bootstrap_base_layout.html.twig

```diff
@@ -0,0 +1,190 @@
+{% use "form_div_layout.html.twig" %}
+
+{# Widgets #}
+
+{% block form_widget_simple -%}
+    {% if type is not defined or type not in ['file', 'hidden'] %}
+        {%- set attr = attr|merge({class: (attr.class|default('') ~ ' form-control')|trim}) -%}
+    {% endif %}
+    {{- parent() -}}
+{%- endblock form_widget_simple %}
```

Comment:
For file inputs, boostrap recommends [switching the class from `.form-control` to `.form-control-file`](https://getbootstrap.com/docs/4.0/components/forms/#form-controls) and this seems to be missing from this theme.
You can include this in by appending '-file' to the default 'form-control' class if the widget type is set to 'file' and slightly changing the type detection condition : 

``` twig
{% block form_widget_simple -%}
    {% if type is not defined or type != 'hidden' %}
        {%- set attr = attr|merge({class: (attr.class|default('') ~ ' form-control' ~ (type|default('') == 'file' ? '-file' : ''))|trim}) -%}
    {% endif %}
    {{- parent() -}}
{%- endblock form_widget_simple %}
```

Reply:
Thank you for the comment.

I've split that widget into a BS 3 and BS 4 variant, and used your suggestion. The layout test for the file input has also been updated.